scipy/optimize/_basinhopping.py

```diff
@@ -682,7 +682,7 @@ def basinhopping(func, x0, niter=100, T=1.0, stepsize=0.5,
     count, i = 0, 0
     message = ["requested number of basinhopping iterations completed"
                " successfully"]
-    for i in range(niter):
+    for i in range(niter+1):
```

Comment:
This change cannot be correct.

Reply:
It's also unrelated to the `maxfev` issue you're addressing. So please remove this second commit.

Reply:
Sorry for my mistake. As it was my first pull request, I somehow messed it up. I have anyhow reversed it. 

Regarding basinhopping, maybe I am wrong. I have still commented my problem with a new example to the original issue #13011 .

Reply:
@prince2110 no worries at all. It's just something to keep in mind: one PR per issue, keeps things easier to review and merge.